packages/yourloops/test/integration/assert/caregivers-list.assert.ts

```diff
@@ -0,0 +1,171 @@
+  expect(within(caregiverRowAfterAdd).getByTestId(`patient-caregivers-table-row-${mockCaregiverUser.username}-button-remove`)).toBeVisible()
+
+  const newCaregiverRow = within(caregiversTableAfterAdd).getByTestId(`patient-caregivers-table-row-${newCaregiverEmail}`)
+  expect(newCaregiverRow).toBeVisible()
+  expect(newCaregiverRow).toHaveTextContent('--new-caregiver@mail.com')
+  const removeCaregiverButton = within(newCaregiverRow).getByTestId(`patient-caregivers-table-row-${newCaregiverEmail}-button-remove`)
+  expect(removeCaregiverButton).toBeVisible()
+
+  await userEvent.click(removeCaregiverButton)
+
+  const removeCaregiverDialog = screen.getByTestId('remove-direct-share-dialog')
+  expect(within(removeCaregiverDialog).getByText('Remove caregiver new-caregiver@mail.com')).toBeVisible()
+  expect(removeCaregiverDialog).toHaveTextContent('Are you sure you want to remove caregiver new-caregiver@mail.com?They will no longer have access to your data.')
+  const cancelButton = within(removeCaregiverDialog).getByRole('button', { name: 'Cancel' })
+  expect(cancelButton).toBeEnabled()
+  const removeButton = within(removeCaregiverDialog).getByRole('button', { name: 'Remove caregiver' })
+  expect(removeButton).toBeEnabled()
+
+  jest.spyOn(NotificationApi, 'getSentInvitations').mockResolvedValueOnce([])
+  jest.spyOn(DirectShareApi, 'getDirectShares').mockResolvedValue([{ status: UserInviteStatus.Accepted, user: mockCaregiverUser }])
+  await userEvent.click(removeButton)
+
+  expect(screen.queryByTestId('remove-direct-share-dialog')).not.toBeInTheDocument()
+  const removeCaregiverSuccessfulSnackbar = screen.getByTestId('alert-snackbar')
+  expect(removeCaregiverSuccessfulSnackbar).toHaveTextContent('Your caregiver has no longer access to your data.')
+  await userEvent.click(within(removeCaregiverSuccessfulSnackbar).getByTitle('Close'))
+
+  const caregiversTableAfterRemove = await screen.findByTestId('patient-caregivers-list')
+  expect(caregiversTableAfterRemove).toBeVisible()
+
+  const caregiverRowAfterRemove = within(caregiversTableAfterRemove).getByTestId(`patient-caregivers-table-row-${mockCaregiverUser.username}`)
+  expect(caregiverRowAfterRemove).toBeVisible()
+  expect(caregiverRowAfterRemove).toHaveTextContent('UserCaregivercaregiver@mail.com')
+  expect(within(caregiverRowAfterRemove).getByTestId(`patient-caregivers-table-row-${mockCaregiverUser.username}-button-remove`)).toBeVisible()
+
+  expect(NotificationApi.cancelInvitation).toHaveBeenCalledWith('my-id', 'target-id', 'new-caregiver@mail.com')
+  expect(within(caregiversTableAfterRemove).queryByTestId(`patient-caregivers-table-row-${newCaregiverEmail}`)).not.toBeInTheDocument()
+}
+
+export const checkAddCaregiverErrorCases = async () => {
+  const addCaregiverButton = await screen.findByTestId('add-caregiver-button')
+  await userEvent.click(addCaregiverButton)
+
+  const addCaregiverDialog = screen.getByTestId('patient-add-caregiver-dialog')
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches to further communicate your intent.  -->
**🎯 suggestion (non-blocking):** ‏
```suggestion
  const addCaregiverDialog = screen.getByRole('dialog')
```